internal/server/data/notify.go

```diff
@@ -92,39 +60,50 @@ func ListenForNotify(ctx context.Context, db *DB, opts ListenForNotifyOptions) (
 
 	listener := &Listener{sqlDB: sqlDB, pgxConn: pgxConn}
 
-	var channel string
-	switch {
-	case opts.GrantsByDestination != "":
-		channel = fmt.Sprintf("grants_%d", opts.OrgID)
-	case opts.DestinationCredentialsByDestinationID != 0:
-		channel = fmt.Sprintf("credreq_%s_%s", opts.OrgID.String(), opts.DestinationCredentialsByDestinationID.String())
-	case opts.DestinationCredentialsByID != 0:
-		channel = fmt.Sprintf("credans_%s_%s", opts.OrgID.String(), opts.DestinationCredentialsByID.String())
-	}
-
-	logging.Debugf("listing for notify on %s", channel)
-	_, err = pgxConn.Exec(ctx, "SELECT listen_on_chan($1)", channel)
-	if err != nil {
-		if err := pgxstdlib.ReleaseConn(sqlDB, pgxConn); err != nil {
-			logging.L.Warn().Err(err).Msgf("release pgx conn")
-		}
-		return nil, err
-	}
-
-	switch {
-	case opts.GrantsByDestination != "":
-		listener.isMatchingNotify = func(payload string) error {
-			var grant grantJSON
-			err := json.Unmarshal([]byte(payload), &grant)
-			if err != nil {
-				return err
-			}
-			destination, _, _ := strings.Cut(grant.Resource, ".")
-			if destination != opts.GrantsByDestination {
-				return errNotificationNoMatch
+	for _, descriptor := range descriptors {
+		_, err = pgxConn.Exec(ctx, "SELECT listen_on_chan($1)", descriptor.Channel())
+		if err != nil {
+			if err := pgxstdlib.ReleaseConn(sqlDB, pgxConn); err != nil {
+				logging.L.Warn().Err(err).Msgf("release pgx conn")
 			}
-			return nil
+			return nil, err
 		}
 	}
 	return listener, nil
 }
+
+type ListenChannelGrantsByDestination struct {
+	OrgID         uid.ID
+	DestinationID uid.ID
+}
+
+func (l ListenChannelGrantsByDestination) Channel() string {
+	return fmt.Sprintf("grantsByDest.%v.%v", l.OrgID, l.DestinationID)
+}
+
+type ListenChannelDestinationCredentialsByDestinationID struct {
+	OrgID         uid.ID
+	DestinationID uid.ID
+}
+
+func (l ListenChannelDestinationCredentialsByDestinationID) Channel() string {
+	return fmt.Sprintf("dcredReq.%v.%v", l.OrgID, l.DestinationID)
```

Comment:
I find these formats a bit confusing. The channel name standard seems to be `abbreviated-name.org-id.destination-id`? Feels like we should document this somewhere.

Reply:
Ya, that's right. I think the `group_membership` should probably be `GroupMembers` to match the convention and reduce the length a bit.

I'll add a comment to the interface to explain the limitations and the convention.

Reply:
I've documented this convention in the godoc for the `ListenChannelDescriptor` interface that these types implement.